tests/preflight/preflight_test.go

```diff
@@ -362,103 +391,66 @@ var _ = Describe("Preflight Tests", func() {
 
 		case imagePullSecFlag:
-			flags += fmt.Sprintf("%s %s ", imagePullSecFlag, val)
+			flags = strings.Join([]string{flags, imagePullSecFlag, val}, " ")
 
 		case serviceAccountFlag:
-			flags += fmt.Sprintf("%s %s ", serviceAccountFlag, val)
+			flags = strings.Join([]string{flags, serviceAccountFlag, val}, " ")
 
 		case cleanupOnFailureFlag:
-			flags += fmt.Sprintf("%s ", cleanupOnFailureFlag)
+			flags = strings.Join([]string{flags, cleanupOnFailureFlag}, " ")
 
 		case logLevelFlag:
-			flags += fmt.Sprintf("%s %s ", logLevelFlag, val)
+			flags = strings.Join([]string{flags, logLevelFlag, val}, " ")
 
 		case kubeconfigFlag:
-			flags += fmt.Sprintf("%s %s ", kubeconfigFlag, val)
+			flags = strings.Join([]string{flags, kubeconfigFlag, val}, " ")
 		}
 	}
 
 	cmd := fmt.Sprintf("%s run %s", preflightBinaryFilePath, flags)
 	log.Infof("Preflight check CMD [%s]", cmd)
-	_, err = shell.RunCmd(cmd)
-	Expect(err).To(BeNil())
+	return shell.RunCmd(cmd)
 }
 
-func runCleanupWithUID(uid string) {
-	cleanDirForFiles(cleanupLogFilePrefix)
-	Expect(err).To(BeNil())
+// Executes cleanup for a particular preflight run
+func runCleanupWithUID(uid string) (*shell.CmdOut, error) {
 	cmd := fmt.Sprintf("%s cleanup --uid %s", preflightBinaryFilePath, uid)
 	log.Infof("Preflight cleanup CMD [%s]", cmd)
-	_, err = shell.RunCmd(cmd)
-	Expect(err).To(BeNil())
+	return shell.RunCmd(cmd)
 }
 
-func runCleanupForAllPreflightResources() {
-	cleanDirForFiles(cleanupLogFilePrefix)
-	Expect(err).To(BeNil())
-	cmd := fmt.Sprintf("%s cleanup -n %s", preflightBinaryFilePath, defaultTestNs)
+// Executes cleanup for all preflight resources
+func runCleanupForAllPreflightResources() (*shell.CmdOut, error) {
+	cmd := fmt.Sprintf("./%s cleanup -n %s", preflightBinaryFilePath, defaultTestNs)
 	log.Infof("Preflight cleanup CMD [%s]", cmd)
-	_, err = shell.RunCmd(cmd)
-	Expect(err).To(BeNil())
-}
-
-func cleanDirForFiles(filePrefix string) {
-	var names []fs.FileInfo
-	names, err = ioutil.ReadDir(preflightBinaryDir)
-	Expect(err).To(BeNil())
-	for _, entry := range names {
-		if strings.HasPrefix(entry.Name(), filePrefix) {
-			err = os.RemoveAll(path.Join([]string{preflightBinaryDir, entry.Name()}...))
-			Expect(err).To(BeNil())
-		}
-	}
-}
-
-func getLogFileData(filePrefix string) ([]byte, error) {
-	var (
-		foundFile   = false
-		logFilename string
-		names       []fs.FileInfo
-	)
-	names, err = ioutil.ReadDir(preflightBinaryDir)
-	Expect(err).To(BeNil())
-	for _, entry := range names {
-		if strings.HasPrefix(entry.Name(), filePrefix) {
-			logFilename = entry.Name()
-			foundFile = true
-			break
-		}
-	}
-
-	if !foundFile {
-		return nil, fmt.Errorf("preflight log file not found")
-	}
-	return ioutil.ReadFile(logFilename)
+	return shell.RunCmd(cmd)
 }
 
+// Individual assertions of successful preflight checks which do not involve any CRUD operations
 func nonCRUDPreflightCheckAssertion(inputFlags map[string]string, outputLog string) {
```

Comment:
except test-cases, move all helper funcs to another file [e.g. utils.go] in same dir - `tests/preflight/`